### ISSUE TITLE
[tune] default metric/mode for reporting in tune

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#10592]
### Describe your feature request

Now that we have 1 default mode/metric, it may be quite nice to be able to do:

```
def run(config):
    score = model.train()
    return score
```
or
```
def run(config):
    score = model.train()
    tune.report(score)
    return score
```
All of these can autoresolve.

### Some proposal

```
def run(config):
    score = model.train()
    # or tune.report(score)
    return score

analysis = tune.run(run, config={}, mode="max")
hp = analysis.get_best_config()

# raises RuntimeError since run doesn't report "accuracy"
analysis = tune.run(run, config={}, metric="accuracy", mode="max")

def run(config):
    score = model.train()
    # or tune.report(score, )
    return {"accuracy": 123}

# raises RuntimeError since you need 'accuracy'
analysis = tune.run(run, config={}, mode="max")

search_alg = Searcher()
analysis = tune.run(run, config={}, search_alg=search_alg, metric="accuracy", mode="max")
hp = analysis.get_best_config() # still works fine
```

**Comment by rkooo567** [comment]
cc @ray-project/ray-tune Can you guys find an owner of this task? I will downgrade P2 otherwise!

**Comment by richardliaw** [comment]
This is fixed.
